### PR TITLE
mimxrt/rt1176: Support for the MIMXRT1170_EVK board.

### PR DESCRIPTION
**Pull request by robert-hh**
The support for the RT1176 MCU and the MIMXRT1176_EVK board is ready since September 2021. It is a common effort of @alphaFred and me.

Even if the RT1176 has two cores, the actual firmware supports only the CM7. We have no good plan how to use the CM4 (see also the discussion about the Portenta H7). The actual MIMXRT1176_EVK port is on par with the existing MIMXRT boards, with the following extensions:

- use 64 MB RAM for the heap.
- support both LAN interfaces as LAN(0) and LAN(1), with LAN(1) being the 1GB interface.

The dual LAN port interface can be adapted as well for the RT1062 mcu. Only I have no 1062 board for testing with 2 LAN ports. 

Being held open and recent, this branch and PR has not a clean commit structure. At some point it was necessary to squash all existing commits into one to be able to rebase it on the ever changing master branch. That is the specific first commit in this PR. The other commits are also just changes to keep the branch recent and working. So they could as well be squashed into one commit.

### CODE REVIEW

**Comment by dpgeorge** [comment]
Thanks for this.  I think the best way to move forward with it is just rebase it and squash all commits into one.

File: ports/mimxrt/mpconfigport.h [hunk 1]
```diff
@@ -29,9 +29,13 @@
 // Board specific definitions
 #include "mpconfigboard.h"
 #include "fsl_common.h"
+#include "lib/nxp_driver/sdk/CMSIS/Include/core_cm7.h"
 
 uint32_t trng_random_u32(void);
 
+// Config level
+#define MICROPY_CONFIG_ROM_LEVEL (MICROPY_CONFIG_ROM_LEVEL_FULL_FEATURES)
```
Comment (dpgeorge) [inline ports/mimxrt/mpconfigport.h:37]: with this enabled, there are a lot of config options in this file that can now be removed, because they are the default (at this full features level)

Comment (dpgeorge) [reply ports/mimxrt/mpconfigport.h:37]: actually, if this change is not needed for this PR to build and work, I'd suggest reverting this line (ie don't add it) and do that in a separate PR, which also cleans up the other config options

File: ports/mimxrt/boards/MIMXRT1170_EVK/qspi_hyper_flash_config.c [hunk 2]
```diff
@@ -0,0 +1,186 @@
+/*
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1170_EVK/qspi_hyper_flash_config.c:1]: Why is this file needed, because I see a common file `hal/qspi_hyper_flash_config.c` that I would think can be reused for this MCU?  If this code has to be different for the RT1170, then can this file be moved to `hal/qspi_hyper_flash_config_MIMXRT1170.c` or something like that?

File: ports/mimxrt/boards/MIMXRT1170_EVK/MIMXRT1170_EVK_flexspi_nor_config.h [hunk 3]
```diff
@@ -0,0 +1,263 @@
+/*
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1170_EVK/MIMXRT1170_EVK_flexspi_nor_config.h:1]: Can this file be put in a common location, like `hal/`?

File: ports/mimxrt/mpconfigport.h (hunk 1 above)
Comment (robert-hh) [reply ports/mimxrt/mpconfigport.h:37]: About the config levels: I did that clean-up at a certain stage for the SAMD port, and it's due for the mimxrt port. But it's better to make it separate.

File: ports/mimxrt/boards/MIMXRT1170_EVK/qspi_hyper_flash_config.c (hunk 2 above)
Comment (robert-hh) [reply ports/mimxrt/boards/MIMXRT1170_EVK/qspi_hyper_flash_config.c:1]: Actually this file is not needed any more, because replaced by the file in hal. So I remove it. Thank you for noticing. This branch was dangling around since Sep 2021, and since then the main mimxrt branch has been cleaned.

File: ports/mimxrt/boards/MIMXRT1170_EVK/MIMXRT1170_EVK_flexspi_nor_config.h (hunk 3 above)
Comment (robert-hh) [reply ports/mimxrt/boards/MIMXRT1170_EVK/MIMXRT1170_EVK_flexspi_nor_config.h:1]: Same as above. Not needed any more.

File: ports/mimxrt/mpconfigport.h (hunk 1 above)
Comment (robert-hh) [reply ports/mimxrt/mpconfigport.h:37]: Actually this line was added in May 2020 by @jimmo, who made the first minimal port for mimxrt. I cannot tell at first glance the side effects of removing it. So I prefer to leave it as it is now and change it in a separate PR, maybe even raising the level to EXTRA_FEATURES. The mimxrt boards all have plenty of flash.

**Comment by robert-hh** [comment]
All commits squashed, unnecessary files removed. 
Compiled and loaded the build with 1176_EVK, 1020_EVK, 1050_EVK, Seeed Arch Mix, Teensy 4.1, Olimex RT1010. MP starts, LAN works, the heap is present.

File: ports/mimxrt/mpconfigport.h (hunk 1 above)
Comment (dpgeorge) [reply ports/mimxrt/mpconfigport.h:37]: OK

**Comment by dpgeorge** [comment]
Rebased and merged in 5e990cc27f86aee07555a5c9188c252108309746 with some minor changes:
- reverted change to esp32 Makefile to set GENERIC_SPIRAM board as default
- fixed "product" entry in board.json for new mimxrt board (it was a copy and paste error, MIMXRT1060_EVK)
- very minor whitespace formatting changes

Thank you both @robert-hh and @alphaFred !

**Comment by robert-hh** [comment]
Thank you for merging and reverting the ESP32 change. Something that sneaked in over time.
